staging/src/k8s.io/apiserver/pkg/storage/interfaces.go

```diff
@@ -166,7 +166,11 @@ type Interface interface {
 
 	// Delete removes the specified key and returns the value that existed at that spot.
 	// If key didn't exist, it will return NotFound storage error.
-	Delete(ctx context.Context, key string, out runtime.Object, preconditions *Preconditions, validateDeletion ValidateObjectFunc) error
+	// If 'suggestion' can contain zero or one element - in such case this can be used as
+	// a suggestion about the current version of the object to avoid read operation from
+	// storage to get it.
+	Delete(ctx context.Context, key string, out runtime.Object, preconditions *Preconditions,
+		validateDeletion ValidateObjectFunc, suggestion ...runtime.Object) error
```

Comment:
Please use `suggestion runtime.Object`.  The existing anti-pattern of using `suggestion ...runtime.Object` makes is unnecessarily hard to find callers that give a suggestion vs. not.  Go does not have optional parameters to functions; please do not try to invent such an approach.

Reply:
We can consider cleaning this up, but I strongly believe consistency is much more important. Given we already have it for Update, as a user I don't want to have two functions that are supposed to have the same semantic and are using a different interface...

Reply:
done

Reply:
done